taskcluster/fenix_taskgraph/target_tasks.py

```diff
@@ -4,36 +4,26 @@
 
 from __future__ import absolute_import, print_function, unicode_literals
 
-import os
-import re
-
 from taskgraph.target_tasks import _target_task, filter_for_tasks_for
 
 
-BETA_SEMVER = re.compile(r'^v\d+\.\d+\.\d+-beta\.\d+$')
-PRODUCTION_SEMVER = re.compile(r'^v\d+\.\d+\.\d+(-rc\.\d+)?$')
-
-
 @_target_task('default')
 def target_tasks_default(full_task_graph, parameters, graph_config):
     """Target the tasks which have indicated they should be run on this project
     via the `run_on_projects` attributes."""
 
     filter = filter_for_tasks_for
     if parameters["tasks_for"] == 'github-release':
```

Comment:
Rather than having a conditional here, I would have some logic in the decision-parameters to set `target_tasks_method` in this case, and just define a new target_tasks_method for handling releases here.

Reply:
How do you set `target_tasks_method` in `decision-parameters`? Is `parameters['target_tasks_method']` a magic value?

Reply:
It isn't magic, but it is looked at by taskgraph generation. In gecko, we set it in the equivalent of `decision-parameters` ([here](https://searchfox.org/mozilla-central/rev/f1e99da78fe6c3c68696358dac06aed90f8112d3/taskcluster/taskgraph/decision.py#308-310), [here](https://searchfox.org/mozilla-central/rev/f1e99da78fe6c3c68696358dac06aed90f8112d3/taskcluster/taskgraph/decision.py#39-118) and [here](https://searchfox.org/mozilla-central/rev/f1e99da78fe6c3c68696358dac06aed90f8112d3/taskcluster/taskgraph/actions/release_promotion.py#268-270,308)). It is used [here](https://searchfox.org/mozilla-central/rev/f1e99da78fe6c3c68696358dac06aed90f8112d3/taskcluster/taskgraph/filter_tasks.py#33-35).

Reply:
Ok, I understand. I didn't realize that `parameters` is used both to contain `taskgraph` parameters (same keys between all projects) and project-specific parameters (arbitrary data that will be ignored by `taskgraph` but used by the project logic elsewhere).

Reply:
I don't like this conditional either and I'm glad you called it out, @tomprince! How can I implement https://searchfox.org/mozilla-central/rev/f1e99da78fe6c3c68696358dac06aed90f8112d3/taskcluster/taskgraph/decision.py#39-118 in fenix? Is there a function I can hook to? If not, I'd rather land the PR as is and make this change in a followup. The reason is: the current patch enables me to publish all parameters for taskgraph-diff. I'd love to land it because it's really valuable in testing my hotfixes. For instance: #5635

Reply:
https://github.com/mozilla-mobile/fenix/pull/5564/files#diff-53e691d50a0b26d43d925c2150567e76R42